cmd/dp-deployer/main.go

```diff
@@ -128,7 +136,7 @@ func initHandlers(ctx context.Context, cfg *config.Configuration, vc *vault.Clie
 	}, nil
 }
 
-func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3) *healthcheck.HealthCheck {
+func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3, nomadHealthClient *health.Client) (*healthcheck.HealthCheck, bool) {
```

Comment:
Instead of a boolean, I'd suggest returning an `error` if a healthcheck fails.
```suggestion
func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3, nomadHealthClient *health.Client) (*healthcheck.HealthCheck, error) {
```
In fact, I'd change this function so it fails early if any heallth check fails. Where this is being used you're `os.Exit`ing on any failure anyway.
```
func startHealthChecks(ctx context.Context, cfg *config.Configuration, vaultChecker *vault.Client, s3sChecker *s3client.S3, s3dChecker *s3client.S3, nomadHealthClient *health.Client) (*healthcheck.HealthCheck, bool) {

	// Create healthcheck object with versionInfo
	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
	if err != nil {
		return nil, errors.Wrap(err, "failed to create service version information")
	}
	hc := healthcheck.New(versionInfo, cfg.HealthcheckCriticalTimeout, cfg.HealthcheckInterval)

	if err := hc.AddCheck("Vault", vaultChecker.Checker); err != nil {
		return nil, errors.Wrap(err, "error adding check for vault")
	}

	if err := hc.AddCheck("S3 secret", s3sChecker.Checker); err != nil {
		return nil, errors.Wrap(err, "error adding check for S3 secrets")
	}

	if err := hc.AddCheck("S3 deployment", s3dChecker.Checker); err != nil {
		return nil, errors.Wrap(err, "error adding check for S3 deployments")
	}

	if err := hc.AddCheck("Nomad", nomadHealthClient.Checker); err != nil {
		return nil, errors.Wrap(err, "error adding check for nomad")
	}

	// Start healthcheck
	hc.Start(ctx)

	return &hc, nil
}
```
for the above to work, you'll need to import `github.com/pkg/errors`